### DISCUSSION TITLE
What are the optimal hardware/devices for running macOS smoothly?

### DISCUSSION THREAD

**Discussion by imthenachoman** [quickemu-project/quickemu#1372] [Quickemu]
I run Linux on my daily driver. I need a new machine and am planning on building a new box.

I want to make sure I can run a macOS VM (hopefully Sonoma) smoothly. 

What kind of hardware should I have? What kind of CPU and how fast? How much RAM? Any specific MB? Any specific GPU or will iGPU work?

**Comment by lj3954** [comment]
> What kind of CPU and how fast?

Virtually any CPU can be used for macOS VMs, since QEMU is capable of emulating several supported models. CPU performance should be nearly that of your host (at least, per core), there's not too much to worry about on that front with KVM. Your motherboard also does not matter whatsoever. macOS VMs should be allocated at least 8GB of RAM for reasonable performance. 

> I want to make sure I can run a macOS VM (hopefully Sonoma) smoothly.

There is no virtualization software capable of virtualizing a GPU with macOS 3D graphics drivers on non-apple hosts. Quickemu does not currently support GPU passthrough. I believe you'd be better suited elsewhere if you need strong graphics performance, with other QEMU frontends such as libvirt/virt-manager. 

Regarding GPUs for a macOS VM, to the best of my knowledge, AMD 5000 series and earlier is the last that's supported. Anything later than that may not work, and NVIDIA is also not a choice. Intel iGPUs from 10th gen and older CPUs can also be used by macOS, AMD iGPUs cannot. Remember, for all of these cases, you'll have to manually set up GPU passthrough and the device will not be able to be used by the host & guest simultaneously.

If you just need macOS here and there for a task that doesn't require GPU acceleration, build whatever system would work best for your needs on the host. I would not recommend buying components specifically for a macOS VM, especially considering you'd have to go back several generations. If macOS support is a high priority to you, unfortunately the only way you're going to be able to get hardware with good support for the operating system into the future is to buy it from Apple. Virtualization won't be an option as Apple finishes transitioning away from the standard off-the-shelf hardware they used for x86_64 machines.

**Reply by imthenachoman** [reply]
Thank you. This is helpful. 

But now I think something is wrong with my current setup because I created a macOS Sonoma VM and it's very choppy/slow. I need to see how much RAM Quickemu allocated.

**Reply by YuvanMichaelVivenzi** [reply]
configured in ``` macos-sonoma.conf ```
